docs/DeveloperGuide.md

```diff
@@ -1,38 +1,416 @@
 # Developer Guide
+## Contents
+* [Acknowledgements](#acknowledgements)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [Manager Component](#manager-component)
+    * [Profile Component](#profile-component)
+    * [TextUI Component](#textui-component)
+    * [Storage Component](#storage-component)
+    * [Parser Component](#parser-component)
+    * [Command Component](#command-component)
+* Implementation
+    * [Add a session](#add-a-session)
+    * [Remove a session](#remove-a-session)
+    * [View a session](#view-a-session)
+    * [List all sessions](#list-sessions)
+    * [Settle a session](#settle-a-session)
+    * [Add an activity](#add-an-activity)
+    * [Remove an activity](#remove-an-activity)
+    * [View an activity](#view-an-activity)
+    * [List all activities](#list-activities)
+    * [Add a group](#add-a-group)
+    * [Remove a group](#remove-a-group)
+    * [View a group](#view-a-group)
+    * [List all groups](#list-groups)
+* [Project Scope](#product-scope)
+* [User Stories](#user-stories)
+* [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+
 
 ## Acknowledgements
+* Address Book (Level-3) - Provide samples of User Guide(UG) and Developer Guide(DG)
+
+## Design
+### Architecture
+![Application Diagram Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/ArchitectureDiagram.drawio.png)
```

Comment:
Perhaps you could label the person and the file icons on the left and right.
![image](https://user-images.githubusercontent.com/58351701/160957050-66a6ad12-6897-4971-8b3c-b550586edc72.png)

Reply:
Is the arrow from manager to textUi correct? Cause the TextUi invokes an instance the manager

Reply:
Is the arrow from manager to textUi correct? Cause the TextUi invokes an instance the manager